src/types/record.ts

```diff
@@ -2,26 +2,52 @@ import { Runtype, Static, create, innerValidate } from '../runtype';
 import { hasKey } from '../util';
 import show from '../show';
 
-type RecordStaticType<O extends { [_: string]: Runtype }, RO extends boolean> = RO extends true
+type RecordStaticType<
+  O extends { [_: string]: Runtype },
+  Part extends boolean,
+  RO extends boolean
+> = Part extends true
+  ? RO extends true
+    ? { readonly [K in keyof O]?: Static<O[K]> }
+    : { [K in keyof O]?: Static<O[K]> }
+  : RO extends true
   ? { readonly [K in keyof O]: Static<O[K]> }
   : { [K in keyof O]: Static<O[K]> };
 
-export interface Record<O extends { [_: string]: Runtype }, RO extends boolean>
-  extends Runtype<RecordStaticType<O, RO>> {
+interface InternalRecord<
```

Comment:
Decided to rename this and maintain the old `Record`` interface.  What do you think?

Also not exporting, I *think* TS would give an error about private dependencies if that were an issue.